### PR TITLE
Implement -femit-local-var-lifetime which adds local (stack) variable…

### PR DESCRIPTION
**Pull request by JohanEngelen**
… lifetime annotation to LLVM IR, which enables sharing stack space for variables whose lifetimes do not overlap.

Resolves issue #2227

This is not enabled by default yet, to prevent miscompilation due to bugs (should be enabled for optimization levels > 0, and when sanitizers are enabled).

Compiling a piece of druntime, this is the stack space diff (output with `-fwarn-stack-size=100`):
```diff
-Without lifetime annotation
+WITH lifetime annotation!
-warning: stack frame size (144) exceeds limit (100) in function '_D3std9algorithm7sorting__T13quickSortImplSQBp3zip10ZipArchive5buildMFNaNfZ9__lambda6TACQDiQBt13ArchiveMemberZQDfMFNaNbNiNfQBlmZv'
+warning: stack frame size (128) exceeds limit (100) in function '_D3std9algorithm7sorting__T13quickSortImplSQBp3zip10ZipArchive5buildMFNaNfZ9__lambda6TACQDiQBt13ArchiveMemberZQDfMFNaNbNiNfQBlmZv'
-warning: stack frame size (128) exceeds limit (100) in function '_D3std7variant__T8VariantNVmi16ZQp__T7handlerTSQBt11concurrency3TidZQBfFEQCtQCs__TQCnVmi16ZQCw4OpIDPG16hPvZl'
+warning: stack frame size (112) exceeds limit (100) in function '_D3std7variant__T8VariantNVmi16ZQp__T7handlerTSQBt11concurrency3TidZQBfFEQCtQCs__TQCnVmi16ZQCw4OpIDPG16hPvZl'
-warning: stack frame size (224) exceeds limit (100) in function '_D3std3uni__T16UnicodeSetParserTSQBf5regex8internal6parser__T6ParserTAyaTSQCuQBpQBmQBg7CodeGenZQBiZQDi8parseSetMFNfZSQElQEk__T13InversionListTSQFlQFk8GcPolicyZQBh'
+warning: stack frame size (192) exceeds limit (100) in function '_D3std3uni__T16UnicodeSetParserTSQBf5regex8internal6parser__T6ParserTAyaTSQCuQBpQBmQBg7CodeGenZQBiZQDi8parseSetMFNfZSQElQEk__T13InversionListTSQFlQFk8GcPolicyZQBh'
-warning: stack frame size (176) exceeds limit (100) in function '_D3std3uni__T16UnicodeSetParserTSQBf5regex8internal6parser__T6ParserTAyaTSQCuQBpQBmQBg7CodeGenZQBiZQDi8parseSetMFZ5applyFNfEQEsQEr__TQEqTQEbZQEy8OperatorKSQFxQFw__T5StackTSQGoQGn__T13InversionListTSQHoQHn8GcPolicyZQBhZQCcZb'
-warning: stack frame size (112) exceeds limit (100) in function '_D3std6string__T8_indexOfTAxaZQoFNaNbNiNfQpwEQBr8typecons__T4FlagVAyaa13_6361736553656e736974697665ZQBoZl'
-warning: stack frame size (160) exceeds limit (100) in function '_D3std6socket10getAddressFNfMAxaMQeZACQBkQBj7Address'
+warning: stack frame size (144) exceeds limit (100) in function '_D3std6socket10getAddressFNfMAxaMQeZACQBkQBj7Address'
-warning: stack frame size (368) exceeds limit (100) in function '_D3std5regex8internal6parser7CodeGen11charsetToIrMFNeSQCa3uni__T13InversionListTSQDbQBb8GcPolicyZQBhZv'
+warning: stack frame size (304) exceeds limit (100) in function '_D3std5regex8internal6parser7CodeGen11charsetToIrMFNeSQCa3uni__T13InversionListTSQDbQBb8GcPolicyZQBhZv'
-warning: stack frame size (144) exceeds limit (100) in function '_D3std5regex8internal6parser7CodeGen13fixRepetitionMFkkkbZv'
+warning: stack frame size (128) exceeds limit (100) in function '_D3std5regex8internal6parser7CodeGen13fixRepetitionMFkkkbZv'
-warning: stack frame size (400) exceeds limit (100) in function '_D3std5regex8internal6parser__T6ParserTAyaTSQBqQBpQBmQBg7CodeGenZQBi11parseEscapeMFNeZv'
+warning: stack frame size (320) exceeds limit (100) in function '_D3std5regex8internal6parser__T6ParserTAyaTSQBqQBpQBmQBg7CodeGenZQBi11parseEscapeMFNeZv'
-warning: stack frame size (704) exceeds limit (100) in function '_D3std5regex8internal9kickstart__T7ShiftOrTaZQl6__ctorMFNcNeKSQCiQChQCe2ir__T5RegexTaZQjAkZSQDmQDlQDiQDc__TQCvTaZQDb'
+warning: stack frame size (576) exceeds limit (100) in function '_D3std5regex8internal9kickstart__T7ShiftOrTaZQl6__ctorMFNcNeKSQCiQChQCe2ir__T5RegexTaZQjAkZSQDmQDlQDiQDc__TQCvTaZQDb'
-warning: stack frame size (432) exceeds limit (100) in function '_D3std5regex8internal12backtracking9CtContext10ctGenGroupMFKAxSQCjQCiQCf2ir8BytecodeiZSQDhQDgQDdQCxQCm7CtState'
+warning: stack frame size (416) exceeds limit (100) in function '_D3std5regex8internal12backtracking9CtContext10ctGenGroupMFKAxSQCjQCiQCf2ir8BytecodeiZSQDhQDgQDdQCxQCm7CtState'
-warning: stack frame size (880) exceeds limit (100) in function '_D3std7process11environment13opIndexAssignFNeNkMNgAaMAxaZANga'
+warning: stack frame size (608) exceeds limit (100) in function '_D3std7process11environment13opIndexAssignFNeNkMNgAaMAxaZANga'
-warning: stack frame size (880) exceeds limit (100) in function '_D3std7process__T15pipeProcessImplSQBhQBg12spawnProcessTAxAaZQBsFNeMQmEQCrQCq8RedirectxHAyaAyaSQDpQDo6ConfigMAxaZSQEiQEh12ProcessPipes'
-warning: stack frame size (880) exceeds limit (100) in function '_D3std7process__T15pipeProcessImplSQBhQBg12spawnProcessTAxaZQBrFNeMQlEQCqQCp8RedirectxHAyaAyaSQDoQDn6ConfigMQCaZSQEhQEg12ProcessPipes'
-warning: stack frame size (896) exceeds limit (100) in function '_D3std7process__T15pipeProcessImplSQBhQBg10spawnShellTAxaTAyaZQBtFNeMQpEQCsQCr8RedirectxHQBfAyaSQDqQDp6ConfigMQCeQCdZSQEmQEl12ProcessPipes'
+warning: stack frame size (608) exceeds limit (100) in function '_D3std7process__T15pipeProcessImplSQBhQBg12spawnProcessTAxAaZQBsFNeMQmEQCrQCq8RedirectxHAyaAyaSQDpQDo6ConfigMAxaZSQEiQEh12ProcessPipes'
+warning: stack frame size (608) exceeds limit (100) in function '_D3std7process__T15pipeProcessImplSQBhQBg12spawnProcessTAxaZQBrFNeMQlEQCqQCp8RedirectxHAyaAyaSQDoQDn6ConfigMQCaZSQEhQEg12ProcessPipes'
+warning: stack frame size (624) exceeds limit (100) in function '_D3std7process__T15pipeProcessImplSQBhQBg10spawnShellTAxaTAyaZQBtFNeMQpEQCsQCr8RedirectxHQBfAyaSQDqQDp6ConfigMQCeQCdZSQEmQEl12ProcessPipes'
-warning: stack frame size (304) exceeds limit (100) in function '_D3std7process__T11executeImplSQBdQBc11pipeProcessTAxAaZQBnFNeQlxHAyaAyaSQCtQCs6ConfigmMAxaZSQDn8typecons__T5TupleTiVQBza6_737461747573TQCsVQCwa6_6f7574707574ZQBz'
-warning: stack frame size (304) exceeds limit (100) in function '_D3std7process__T11executeImplSQBdQBc11pipeProcessTAxaZQBmFNeQkxHAyaAyaSQCsQCr6ConfigmMQBkZSQDm8typecons__T5TupleTiVQBza6_737461747573TQCsVQCwa6_6f7574707574ZQBz'
-warning: stack frame size (320) exceeds limit (100) in function '_D3std7process__T11executeImplS_DQBfQBe9pipeShellFNfMAxaEQCdQCc8RedirectxHAyaAyaSQDbQDa6ConfigMQBqQyZSQDwQDv12ProcessPipesTQCsTQCbZQEkFNeQDgxQCqQCmmMQDsQDaZSQFz8typecons__T5TupleTiVQEda6_737461747573TQEwVQFaa6_6f7574707574ZQBz'
+warning: stack frame size (224) exceeds limit (100) in function '_D3std7process__T11executeImplSQBdQBc11pipeProcessTAxAaZQBnFNeQlxHAyaAyaSQCtQCs6ConfigmMAxaZSQDn8typecons__T5TupleTiVQBza6_737461747573TQCsVQCwa6_6f7574707574ZQBz'
+warning: stack frame size (224) exceeds limit (100) in function '_D3std7process__T11executeImplSQBdQBc11pipeProcessTAxaZQBmFNeQkxHAyaAyaSQCsQCr6ConfigmMQBkZSQDm8typecons__T5TupleTiVQBza6_737461747573TQCsVQCwa6_6f7574707574ZQBz'
+warning: stack frame size (240) exceeds limit (100) in function '_D3std7process__T11executeImplS_DQBfQBe9pipeShellFNfMAxaEQCdQCc8RedirectxHAyaAyaSQDbQDa6ConfigMQBqQyZSQDwQDv12ProcessPipesTQCsTQCbZQEkFNeQDgxQCqQCmmMQDsQDaZSQFz8typecons__T5TupleTiVQEda6_737461747573TQEwVQFaa6_6f7574707574ZQBz'
-warning: stack frame size (224) exceeds limit (100) in function '_D3std4path__T9buildPathTAAxaZQqFNaNbNfMQpZAya'
+warning: stack frame size (208) exceeds limit (100) in function '_D3std4path__T9buildPathTAAxaZQqFNaNbNfMQpZAya'
-warning: stack frame size (1360) exceeds limit (100) in function '_D3std3net4curl4HTTP4Impl15onReceiveHeaderMFNdDFIAaIQdZvZ9__lambda2MFIQvZv'
+warning: stack frame size (1152) exceeds limit (100) in function '_D3std3net4curl4HTTP4Impl15onReceiveHeaderMFNdDFIAaIQdZvZ9__lambda2MFIQvZv'
-warning: stack frame size (240) exceeds limit (100) in function '_D3std4json__T6toJSONTSQv5array__T8AppenderTAyaZQoZQBlFKQBiKxSQCiQCh9JSONValueIbIEQDcQDb11JSONOptionsZ11toValueImplMFNeKxQCimZv'
+warning: stack frame size (224) exceeds limit (100) in function '_D3std4json__T6toJSONTSQv5array__T8AppenderTAyaZQoZQBlFKQBiKxSQCiQCh9JSONValueIbIEQDcQDb11JSONOptionsZ11toValueImplMFNeKxQCimZv'
-warning: stack frame size (288) exceeds limit (100) in function '_D3std8internal4math11biguintcore11mulInternalFNaNbNfAkAxkQdZv'
+warning: stack frame size (224) exceeds limit (100) in function '_D3std8internal4math11biguintcore11mulInternalFNaNbNfAkAxkQdZv'
-warning: stack frame size (176) exceeds limit (100) in function '_D3std6format5write__T14formattedWriteTSQBmQBl8NoOpSinkTaTmTmZQBoFNaNfQBfMxAammZk'
+warning: stack frame size (192) exceeds limit (100) in function '_D3std6format5write__T14formattedWriteTSQBmQBl8NoOpSinkTaTmTmZQBoFNaNfQBfMxAammZk'
-warning: stack frame size (176) exceeds limit (100) in function '_D3std6format4spec__T10FormatSpecTaZQp6fillUpMFNaNlNfZv'
+warning: stack frame size (144) exceeds limit (100) in function '_D3std6format4spec__T10FormatSpecTaZQp6fillUpMFNaNlNfZv'
-warning: stack frame size (240) exceeds limit (100) in function '_D3std6format8internal5write__T15formatValueImplTSQBw5array__T8AppenderTAyaZQoTPvTaZQCbFNaNfKQBsMxPvMKxSQDyQDx4spec__T10FormatSpecTaZQpZv'
+warning: stack frame size (160) exceeds limit (100) in function '_D3std6format8internal5write__T15formatValueImplTSQBw5array__T8AppenderTAyaZQoTPvTaZQCbFNaNfKQBsMxPvMKxSQDyQDx4spec__T10FormatSpecTaZQpZv'
-warning: stack frame size (352) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5array__T8AppenderTAyaZQoTQhTaZQBxFNaNfKQBsKQzMKxSQDtQDs4spec__T10FormatSpecTaZQpZv'
+warning: stack frame size (272) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5array__T8AppenderTAyaZQoTQhTaZQBxFNaNfKQBsKQzMKxSQDtQDs4spec__T10FormatSpecTaZQpZv'
-warning: stack frame size (352) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5stdio4File17LockingTextWriterTAyaTaZQCdFNfKQBwKQrMKxSQDxQDw4spec__T10FormatSpecTaZQpZv'
+warning: stack frame size (272) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5stdio4File17LockingTextWriterTAyaTaZQCdFNfKQBwKQrMKxSQDxQDw4spec__T10FormatSpecTaZQpZv'
-warning: stack frame size (240) exceeds limit (100) in function '_D3std6format8internal5write__T15formatValueImplTSQBw5array__T8AppenderTAyaZQoTPxSQDcQDb4spec__T10FormatSpecTaZQpTaZQDhFNaNfKQCyMxPQByMKxQCeZv'
+warning: stack frame size (160) exceeds limit (100) in function '_D3std6format8internal5write__T15formatValueImplTSQBw5array__T8AppenderTAyaZQoTPxSQDcQDb4spec__T10FormatSpecTaZQpTaZQDhFNaNfKQCyMxPQByMKxQCeZv'
-warning: stack frame size (352) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5array__T8AppenderTAyaZQoTAxaTaZQByFNaNfKQBtKQtMKxSQDuQDt4spec__T10FormatSpecTaZQpZv'
+warning: stack frame size (272) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5array__T8AppenderTAyaZQoTAxaTaZQByFNaNfKQBtKQtMKxSQDuQDt4spec__T10FormatSpecTaZQpZv'
-warning: stack frame size (352) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5array__T8AppenderTyAaZQoTAyaTaZQByFNaNfKQBtKQtMKxSQDuQDt4spec__T10FormatSpecTaZQpZv'
+warning: stack frame size (272) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5array__T8AppenderTyAaZQoTAyaTaZQByFNaNfKQBtKQtMKxSQDuQDt4spec__T10FormatSpecTaZQpZv'
-warning: stack frame size (352) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5stdio4File17LockingTextWriterTAxaTaZQCdFNfKQBwKQrMKxSQDxQDw4spec__T10FormatSpecTaZQpZv'
+warning: stack frame size (272) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5stdio4File17LockingTextWriterTAxaTaZQCdFNfKQBwKQrMKxSQDxQDw4spec__T10FormatSpecTaZQpZv'
-warning: stack frame size (240) exceeds limit (100) in function '_D3std6format8internal5write__T15formatValueImplTSQBw5array__T8AppenderTAyaZQoTPxhTaZQCcFNaNfKQBtMxPhMKxSQDzQDy4spec__T10FormatSpecTaZQpZv'
+warning: stack frame size (160) exceeds limit (100) in function '_D3std6format8internal5write__T15formatValueImplTSQBw5array__T8AppenderTAyaZQoTPxhTaZQCcFNaNfKQBtMxPhMKxSQDzQDy4spec__T10FormatSpecTaZQpZv'
-warning: stack frame size (256) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5array__T8AppenderTAyaZQoTAyAaTaZQBzFNaNfKQBuKQuMKxSQDvQDu4spec__T10FormatSpecTaZQpZv'
+warning: stack frame size (240) exceeds limit (100) in function '_D3std6format8internal5write__T11formatRangeTSQBs5array__T8AppenderTAyaZQoTAyAaTaZQBzFNaNfKQBuKQuMKxSQDvQDu4spec__T10FormatSpecTaZQpZv'
-warning: stack frame size (240) exceeds limit (100) in function '_D3std6format8internal5write__T15formatValueImplTSQBw5array__T8AppenderTAyaZQoTPSQDb11parallelism12AbstractTaskTaZQDfFNaNfKQCwMxPSQEyQBxQBnMKxSQFlQFk4spec__T10FormatSpecTaZQpZv'
+warning: stack frame size (160) exceeds limit (100) in function '_D3std6format8internal5write__T15formatValueImplTSQBw5array__T8AppenderTAyaZQoTPSQDb11parallelism12AbstractTaskTaZQDfFNaNfKQCwMxPSQEyQBxQBnMKxSQFlQFk4spec__T10FormatSpecTaZQpZv'
-warning: stack frame size (448) exceeds limit (100) in function '_D3std4file8copyImplFNeMAxaMQeMPxaMQeEQBk8typecons__T4FlagVAyaa18_707265736572766541747472696275746573ZQByZv'
+warning: stack frame size (432) exceeds limit (100) in function '_D3std4file8copyImplFNeMAxaMQeMPxaMQeEQBk8typecons__T4FlagVAyaa18_707265736572766541747472696275746573ZQByZv'
-warning: stack frame size (592) exceeds limit (100) in function '_D3std4file7tempDirFNeZ__T15findExistingDirTAyaTQeTQhTQkTQnTQqZQBlFNfLQBaLQBeLQBiLQBmLQBqLQBuZQBy'
+warning: stack frame size (512) exceeds limit (100) in function '_D3std4file7tempDirFNeZ__T15findExistingDirTAyaTQeTQhTQkTQnTQqZQBlFNfLQBaLQBeLQBiLQBmLQBqLQBuZQBy'
-warning: stack frame size (144) exceeds limit (100) in function '_D3std12experimental9allocator15building_blocks14allocator_list__T13AllocatorListTSQDdQDcQCr8showcase14mmapRegionListFmZ7FactoryTSQEyQExQEmQEf14null_allocator13NullAllocatorZQEe12addAllocatorMFNaNbNimZPSQHtQHsQHhQHaQGm__TQFzTQFnTQDwZQGl4Node'
+warning: stack frame size (112) exceeds limit (100) in function '_D3std12experimental9allocator15building_blocks14allocator_list__T13AllocatorListTSQDdQDcQCr8showcase14mmapRegionListFmZ7FactoryTSQEyQExQEmQEf14null_allocator13NullAllocatorZQEe12addAllocatorMFNaNbNimZPSQHtQHsQHhQHaQGm__TQFzTQFnTQDwZQGl4Node'
-warning: stack frame size (160) exceeds limit (100) in function '_D3std8encoding14EncodingScheme6createFAyaZ24registerDefaultEncodingsFZb'
+warning: stack frame size (240) exceeds limit (100) in function '_D3std8encoding14EncodingScheme6createFAyaZ24registerDefaultEncodingsFZb'
-warning: stack frame size (544) exceeds limit (100) in function '_D3std8datetime8timezone13PosixTimeZone11getTimeZoneFNeAyaQdZyCQCjQCiQCcQBw'
+warning: stack frame size (528) exceeds limit (100) in function '_D3std8datetime8timezone13PosixTimeZone11getTimeZoneFNeAyaQdZyCQCjQCiQCcQBw'
-warning: stack frame size (192) exceeds limit (100) in function '_D3std11concurrency10MessageBox3putMFKSQBlQBk7MessageZv'
+warning: stack frame size (160) exceeds limit (100) in function '_D3std11concurrency10MessageBox3putMFKSQBlQBk7MessageZv'
-warning: stack frame size (192) exceeds limit (100) in function '_D3std9algorithm7sorting__T13quickSortImplSQBp10functional__T9binaryFunVAyaa17_612e74696d6554203c20622e74696d6554VQBqa1_61VQBza1_62ZQCtTASQFg8datetime8timezone13PosixTimeZone14TempTransitionZQGiFNaNbNiNfQCpmZv'
+warning: stack frame size (176) exceeds limit (100) in function '_D3std9algorithm7sorting__T13quickSortImplSQBp10functional__T9binaryFunVAyaa17_612e74696d6554203c20622e74696d6554VQBqa1_61VQBza1_62ZQCtTASQFg8datetime8timezone13PosixTimeZone14TempTransitionZQGiFNaNbNiNfQCpmZv'
-warning: stack frame size (160) exceeds limit (100) in function '_D3std9algorithm7sorting__T13quickSortImplSQBp10functional__T9binaryFunVAyaa17_612e74696d6554203c20622e74696d6554VQBqa1_61VQBza1_62ZQCtTASQFg8datetime8timezone13PosixTimeZone10LeapSecondZQGeFNaNbNiNfQClmZv'
+warning: stack frame size (144) exceeds limit (100) in function '_D3std9algorithm7sorting__T13quickSortImplSQBp10functional__T9binaryFunVAyaa17_612e74696d6554203c20622e74696d6554VQBqa1_61VQBza1_62ZQCtTASQFg8datetime8timezone13PosixTimeZone10LeapSecondZQGeFNaNbNiNfQClmZv'
-warning: stack frame size (160) exceeds limit (100) in function '_D3std9algorithm7sorting__T13quickSortImplSQBp10functional__T9binaryFunVAyaa5_61203c2062VQra1_61VQza1_62ZQBsTAQBmZQDjFNaNbNiNfQrmZv'
+warning: stack frame size (144) exceeds limit (100) in function '_D3std9algorithm7sorting__T13quickSortImplSQBp10functional__T9binaryFunVAyaa5_61203c2062VQra1_61VQza1_62ZQBsTAQBmZQDjFNaNbNiNfQrmZv'
```

### CODE REVIEW

File: runtime/CMakeLists.txt [hunk 1]
```diff
@@ -38,7 +38,7 @@ set(BUILD_SHARED_LIBS     AUTO                                CACHE STRING "Whet
 set(D_FLAGS               -w;-de;-preview=dip1000;-preview=dtorfields;-preview=fieldwise CACHE STRING "Runtime D compiler flags, separated by ';'")
 set(D_EXTRA_FLAGS         ""                                  CACHE STRING "Runtime extra D compiler flags, separated by ';'")
 set(D_FLAGS_DEBUG         -g;-link-defaultlib-debug;-d-debug  CACHE STRING "Runtime D compiler flags (debug libraries), separated by ';'")
-set(D_FLAGS_RELEASE       -O3;-release                        CACHE STRING "Runtime D compiler flags (release libraries), separated by ';'")
+set(D_FLAGS_RELEASE       -O3;-release;-femit-local-var-lifetime  CACHE STRING "Runtime D compiler flags (release libraries), separated by ';'")
```
Comment (JohanEngelen) [inline runtime/CMakeLists.txt:41]: Although disabled by default for user code, for druntime I want to enable it, but let's see what the CI testing says.

**Comment by JohanEngelen** [comment]
frustrating... I cannot locally reproduce the crash/assert seen on the buildservers... :(

**Comment by JohanEngelen** [comment]
Link to PDF with more stack_use_after_scope tests, pointing to other lifetime marks (likely requiring more implementation work in LDC).
https://llvm.org/devmtg/2016-11/Slides/Buka-StackUseAfterScope.pdf

**Comment by JohanEngelen** [comment]
Great, it's working now. Test fails due to bug in phobos:
https://github.com/ldc-developers/phobos/blob/6c83b490f7d6c66bf430e5249dae608848d3ac2c/std/typecons.d#LL7088C1-L7108C46
```d
pure @system unittest
{
    foreach (MyRefCounted; AliasSeq!(SafeRefCounted, RefCounted))
    {
        MyRefCounted!int* p;
        {
            auto rc1 = MyRefCounted!int(5);
            p = &rc1; // assignment of scoped variable
            assert(rc1 == 5);
            assert(rc1._refCounted._store._count == 1);
            auto rc2 = rc1;
            assert(rc1._refCounted._store._count == 2);
            // Reference semantics
            rc2 = 42;
            assert(rc1 == 42);
            rc2 = rc2;
            assert(rc2._refCounted._store._count == 2);
            rc1 = rc2;
            assert(rc1._refCounted._store._count == 2);
        }
        assert(p._refCounted._store == null);  // use after scope!
```

**Comment by kinke** [comment]
Great! - The `core.thread.fiber` failure on macOS-arm64 seems to be sporadic (with shared libs only?). That test is disabled for Linux-aarch64 (https://github.com/ldc-developers/ldc/blob/aa64fad44b8ad926af97d019f4ef68f6f5636825/.cirrus.yml#L61), so probably needs the same hack on macOS for now.

**Comment by JohanEngelen** [comment]
Good to merge?

**Pull request merged**